packages/react-instantsearch/src/core/createInstantSearchManager.js

```diff
@@ -41,6 +41,11 @@ export default function createInstantSearchManager({
     searching: false,
   });
 
+  function updateClient(client) {
+    helper.setClient(client);
+    search(); // Will it work?
```

Comment:
Yes we should trigger a new search when client and indexName are updated because both can have an impact on the result set:
- changing the index (obvious one)
- updating the client: going from logged off to logged in and having a dedicated API key that will update the result set.

Works?